net/dns/resolved.go

```diff
@@ -160,26 +253,25 @@ func (m *resolvedManager) resync(ctx context.Context, signals chan *dbus.Signal)
 			// The resolved bus name has a new owner, meaning resolved
 			// restarted. Reprogram current config.
 			m.logf("systemd-resolved restarted, syncing DNS config")
-			m.mu.Lock()
-			err := m.syncLocked(ctx)
+			err := m.sync(ctx, r1Manager, lastConfig)
 			// Set health while holding the lock, because this will
 			// graciously serialize the resync's health outcome with a
 			// concurrent SetDNS call.
 			health.SetDNSOSHealth(err)
-			m.mu.Unlock()
 			if err != nil {
 				m.logf("failed to configure systemd-resolved: %v", err)
 			}
 		}
 	}
 }
 
-func (m *resolvedManager) syncLocked(ctx context.Context) error {
+// sync is only called from the run goroutine to sync.
```

Comment:
once you have a comment at all, the comment should be in the normal form to start, and then you can end with this info.

So start with saying what sync _does_.

But while I'm looking at the body, I'm thinking `setConfigOverDBus` or something would be a better name.